src/GraphODataTemplateWriter/Settings/ConfigurationService.cs

```diff
@@ -66,39 +66,12 @@ private static TemplateWriterSettings LoadSettingsForLanguage()
 
             TemplateWriterSettings.mainSettingsObject = mainTWS;
             
-            //First dynamically create a new class that holds settings for the target language
-            //We store a reference on the default constructor to the mainTWS and then copy
-            //all properties on it.
-
-            var targetLanguageTypeName = mainTWS.TargetLanguage + "Settings";
-            var targetLanguageAN = new AssemblyName(targetLanguageTypeName);
-            AssemblyBuilder assemblyBuilder = AppDomain.CurrentDomain.DefineDynamicAssembly(targetLanguageAN, AssemblyBuilderAccess.Run);
-            ModuleBuilder moduleBuilder = assemblyBuilder.DefineDynamicModule("MainModule");
-            TypeBuilder tb = moduleBuilder.DefineType(targetLanguageTypeName
-                                , TypeAttributes.Public
-                                | TypeAttributes.Class
-                                | TypeAttributes.AutoClass
-                                | TypeAttributes.BeforeFieldInit
-                                | TypeAttributes.AutoLayout
-                                , typeof(TemplateWriterSettings));
-
-            ConstructorBuilder ctor = tb.DefineConstructor(MethodAttributes.Public, CallingConventions.Standard, Type.EmptyTypes);
-
-            ILGenerator ctorIL = ctor.GetILGenerator();
-
-            ctorIL.Emit(OpCodes.Ldarg_0);
-            ctorIL.Emit(OpCodes.Call, typeof(TemplateWriterSettings).GetConstructor(Type.EmptyTypes));
-            ctorIL.Emit(OpCodes.Ldarg_0);
-            ctorIL.Emit(OpCodes.Call, typeof(TemplateWriterSettings).GetMethod("CopyPropertiesFromMainSettings"));
-            ctorIL.Emit(OpCodes.Ret);
-
-            Type targetLanguageType = tb.CreateType();
-
             //Call the generic GetConfiguration method with our new type.
-            return (TemplateWriterSettings)typeof(IConfigurationProvider)
-                                            .GetMethod("GetConfiguration")
-                                            .MakeGenericMethod(targetLanguageType)
-                                            .Invoke(_configurationProvider, new object[] { });
```

Comment:
Yes! Thank you! You are ticking off boxes from my todo list w.r.t engineering debt!